vnext/Microsoft.ReactNative/Views/Image/ReactImage.cpp

```diff
@@ -128,6 +128,33 @@ winrt::IAsyncOperation<winrt::InMemoryRandomAccessStream> ReactImage::GetImageMe
       co_return nullptr;
   }
 }
+template <typename TImage>
+std::wstring GetUriFromImage(const TImage &image) {
+  return image.UriSource().ToString().c_str();
+}
+template <>
+std::wstring GetUriFromImage(const winrt::Uri &uri) {
+  return uri != nullptr ? uri.ToString().c_str() : L"<no Uri available>";
+}
+
+template <typename TImage>
+void ImageFailed(const TImage &image, const xaml::ExceptionRoutedEventArgs &args) {
+  cwdebug << L"Failed to load image " << GetUriFromImage(image) << L" (" << args.ErrorMessage().c_str() << L")"
+          << std::endl;
+}
+
+// TSourceFailedEventArgs can be either LoadedImageSourceLoadCompletedEventArgs or
+// SvgImageSourceFailedEventArgs, because they both have Status() properties
+// and the type of status are both enums with the same meaning
+// See LoadedImageSourceLoadStatus and SvgImageSourceLoadStatus.
+template <typename TImage, typename TSourceFailedEventArgs>
+void ImageFailed(const TImage &image, const TSourceFailedEventArgs &args) {
+  // https://docs.microsoft.com/en-us/uwp/api/windows.ui.xaml.media.loadedimagesourceloadstatus
+  constexpr std::wstring_view statusNames[] = {L"Success", L"NetworkError", L"InvalidFormat", L"Other"};
+  assert((int)args.Status() < ARRAYSIZE(statusNames));
```

Comment:
>Status [](start = 19, length = 6)

Since this is template and we can't guarantee the type is a unsinged version, shouldn't we also check for >= 0